ts/javascript/prettier.ts

```diff
@@ -0,0 +1,89 @@
+import { Check } from "../check.ts";
+import { Executable } from "../executable.ts";
+import { escapeShellArg, nixRaw, nixStrLit } from "../nix.ts";
+import { Package } from "../package.ts";
+import { Plugin } from "../project.ts";
+
+const mkConfigFile = (config: Record<string, unknown>) =>
+  nixRaw`(pkgs.writeText "prettier-config" ${nixStrLit(
+    JSON.stringify(config),
+  )})`;
+
+/**
+ * A garn plugin that adds a `format` `Executable` that formats your code using
+ * prettier, and a `Check` to verify that your code is formatted.
+ *
+ * @param opts.glob An optional glob pattern to limit what to format/check
+ * @param opts.config An optional prettier configuration
+ *
+ * It will use the version of prettier defined in your package.json, otherwise
+ * if that is not configured, it will fallback to using prettier from nixpkgs.
+ *
+ * Example:
+ * ```typescript
+ * export const myProject = mkNpmProject({ ... })
+ *   .add(garn.javascript.prettier({
+ *     glob: "./src",
```

Comment:
This is probably relative to the `src` of the project, right? Maybe this should be mentioned?